File: haskell/gigasecond/Gigasecond.hs
module Gigasecond (fromDay) where

import Data.Time.Calendar (Day, addDays)

fromDay :: Day -> Day
fromDay = addDays gigasecondInDays

gigasecondInDays :: Integer
gigasecondInDays = 10^9 `div` 60 `div` 60 `div` 24

File: go/gigasecond/gigasecond.go
package gigasecond

import "time"

// A Gigasecond is 10^9 seconds.
const Gigasecond = 1e9 * time.Second

// AddGigasecond returns the time (t + Gigasecond).
func AddGigasecond(t time.Time) time.Time {
	return t.Add(Gigasecond)
}
